File: Backend/src/main/java/com/example/riskfactors/model/SPF.java
package com.example.riskfactors.model;

import lombok.*;
import javax.persistence.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.SequenceGenerator;

@Setter
@Getter
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class SPF {

    @Id
    @SequenceGenerator(
            name = "SPFSequence",
            sequenceName = "SPFSequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "SPFSequence"
    )
    private Long SPFId;
    private boolean isSpfRecordMissing = true;
    private boolean isSpfRecordMalformed = true;
    private boolean doesSpfRecordContainsWildcard = false;
    private boolean doesspfRecordContainsaSoftfail = false;
    private String target;
}

File: Frontend/src/app/business-logic/SPF.ts

export class SPF {
    public isSpfRecordMissing:boolean = true;
    public isSpfRecordMalformed:boolean = true;
    public doesSpfRecordContainsWildcard:boolean = false;
    public doesspfRecordContainsaSoftfail:boolean = false;
    public target:String | undefined;
}
